micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/ExecutorServiceMetrics.java

```diff
@@ -61,6 +64,8 @@ public class ExecutorServiceMetrics implements MeterBinder {
 
     private static final String DESCRIPTION_POOL_SIZE = "The current number of threads in the pool";
 
+    private final Set<Meter.Id> registeredMeterIds = ConcurrentHashMap.newKeySet();
```

Comment:
I don't think we need to use the Concurrent variant here but it also shouldn't matter much if we do.